hudi-client/hudi-client-common/src/test/java/org/apache/hudi/utils/HoodieWriterClientTestHarness.java

```diff
@@ -551,7 +565,7 @@ protected List<HoodieRecord<RawTripTestPayload>> dedupForCopyOnWriteStorage(Hood
     int dedupParallelism = records.getNumPartitions() + additionalParallelism;
     BaseHoodieWriteClient writeClient = getHoodieWriteClient(writeConfig);
     HoodieReaderContext readerContext = writeClient.getEngineContext()
-        .getReaderContextFactoryForWrite(metaClient, HoodieRecord.HoodieRecordType.AVRO, writeConfig.getProps()).getContext();
+        .getReaderContextFactoryForWrite(metaClient, HoodieRecord.HoodieRecordType.AVRO, writeConfig.getProps(), true).getContext();
```

Comment:
all these places where we are setting to true for `outputsCustomPayloads` will be fixed in your next patch where we will be looking to move away from payload based HoodieRecord ?

Reply:
Yes, the decision will be based on the payload class name set and handled internally in the context